### PR TITLE
[Installer] Add documentation for installation options

### PR DESCRIPTION
**Pull request by amorde**
Closes https://github.com/CocoaPods/CocoaPods/issues/8198
Closes https://github.com/CocoaPods/guides.cocoapods.org/issues/142

I took an initial pass at this with the knowledge I have, but could use some help verifying these

### CODE REVIEW

File: lib/cocoapods/installer/installation_options.rb [hunk 1]
```diff
@@ -101,14 +101,73 @@ def hash
         to_h.hash
       end
 
+      # Whether to clean the sources of the pods during installation
+      #
+      # Cleaning removes any files not used by the pod as specified by the podspec and the platforms
+      # that the project supports
```
[the remark this answers is not in the document]
Comment (paulb777) [reply lib/cocoapods/installer/installation_options.rb]: Is it the same as deintegrate? Can it say, "Cleaning removes previous CocoaPods installations from the project before starting the installation" ?

[the remark this answers is not in the document]
Comment (amorde) [reply lib/cocoapods/installer/installation_options.rb]: hmm I think you're right, which makes way more sense than what I have written. I was going off some of the comments in the code

[the remark this answers is not in the document]
Comment (dnkoutso) [reply lib/cocoapods/installer/installation_options.rb]: I could have sworn this clean means to clean the sandbox for all pods being installed. I.e this means the Pods/Headers will be removed etc.

[the remark this answers is not in the document]
Comment (amorde) [reply lib/cocoapods/installer/installation_options.rb]: I'll look into this one some more when I get the chance

**Comment by amorde** [comment]
@segiddins do you know if this page uses the function docs? I was under the impression it does

**Comment by amorde** [comment]
@CocoaPodsBot merge on green

**Comment by amorde** [comment]
Darn, bot must be sleeping

**Pull request merged**